extension/src/popup/helpers/account.ts

```diff
@@ -15,3 +22,94 @@ export const sortBalances = (balances: Balances) => {
 
   return collection;
 };
+
+export const getIsPayment = (type: Horizon.OperationResponseType) =>
+  [
+    Horizon.OperationResponseType.payment,
+    Horizon.OperationResponseType.pathPayment,
+    Horizon.OperationResponseType.pathPaymentStrictSend,
+  ].includes(type);
+
+export const getIsSwap = (operation: HorizonOperation) =>
+  operation.type_i === 13 && operation.source_account === operation.to;
+
+interface SortOperationsByAsset {
+  operations: Array<HorizonOperation>;
+  balances: Array<Types.AssetBalance | Types.NativeBalance>;
+}
+
+export interface AssetOperations {
+  [key: string]: Array<HorizonOperation>;
+}
+
+export const sortOperationsByAsset = ({
+  balances,
+  operations,
+}: SortOperationsByAsset) => {
+  const assetOperationMap = {} as AssetOperations;
+  balances.forEach((bal) => {
```

Comment:
I think this might break w/ the LP shares PR I just merged 🤔  since it woun't have the `token` field :/ 

I think we can handle by if the balance doesn't have a `token` field, we can ignore it. Ie ignoring LP shares for here

Reply:
oh actually looks like the map should include it. So adding the key as the LP identifier to the AssetOperations map, and then adding the op to it